packages/scaffold-config/src/ct-detect-third-party.ts

```diff
@@ -46,14 +87,46 @@ export async function detectThirdPartyCTFrameworks (
   projectRoot: string,
 ): Promise<Cypress.ThirdPartyComponentFrameworkDefinition[]> {
   try {
-    const fullPathGlobs = [
-      path.join(projectRoot, CT_FRAMEWORK_GLOBAL_GLOB),
-      path.join(projectRoot, CT_FRAMEWORK_NAMESPACED_GLOB),
-    ].map((x) => x.replaceAll('\\', '/'))
+    let fullPathGlobs
+    let packageJsonPaths: string[] = []
+
+    // Start at the project root and check each directory above it until we see
+    // an indication that the current directory is the root of the repository.
+    await findUp(async (directory: string) => {
+      fullPathGlobs = [
+        path.join(directory, CT_FRAMEWORK_GLOBAL_GLOB),
+        path.join(directory, CT_FRAMEWORK_NAMESPACED_GLOB),
+      ].map((x) => x.replaceAll('\\', '/'))
+
+      debug('searching for third-party dependencies with globs %o', fullPathGlobs)
+
+      const newPackagePaths = await globby(fullPathGlobs)
+
+      if (newPackagePaths.length > 0) {
+        debug('found third-party dependencies %o', newPackagePaths)
+      }
+
+      packageJsonPaths = [...packageJsonPaths, ...newPackagePaths]
+
+      const isCurrentRepositoryRoot = await isRepositoryRoot(directory)
```

Comment:
I can give this PR a test later, but definitely will be good to see some tests around this full functionality (whether it's Cy-in-Cy, or just an elaborate unit/integration test in this package).

Reply:
I added a unit test in this package, let me know what you think